src/io_handler.c

```diff
@@ -67,13 +92,22 @@ int handle_root_filename(char * input_file, int * run_no, double * beam_energy)
     return 0;
 }
 
+int handle_root_filename_simul(char * input_file, double * beam_energy) {
+    int chk = check_root_filename(input_file);
+    if (chk) return chk;
+
+    // Beam energy of simulation is fixed here!
+    *beam_energy = 11.;
+    return 0;
+}
```

Comment:
Creating a new method with mostly the same code as the old one is not good practice. Maybe you can better implement this with smart argument handling and overloading? It's not super necessary, but I'd rather keep the code as clean as possible to avoid future sneaky bugs.

Reply:
Fixed! used overload as my weapon of choice.